### PR TITLE
lesson 2 task

### PR DESCRIPTION
**Pull request by AZhurav1ev**
Added:
1. Popup on "Profile" button;
2. Form verification function with confirm and warning messages;
3. Form clearing function;

Fixed chaining of promises and confirm message. Added types to inputs and improved rating check function.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -9,13 +9,42 @@ const small_film_set = [
...
+        webix.message({
+            text: `Validation is successful`,
+            type: "success",
+            expire: 1000
+        });
+    }
+}
+
+function clearValidation() {
+    webix.confirm(`Do you realy want to clear validation?`)
```
Comment (Zwillinge) [inline script.js]: (1) why chaining promises for sync code? `.clearValidation` and `.clear` should be placed into one and the same handler.

https://medium.com/@fagnerbrack/promises-sync-code-disaster-e9d41a3c7279

"Besides the obvious performance implications of abusing the Promises chain by running synchronous functions unnecessarily, you are also breaking the reader’s expectation by changing the intent of how the code is supposed to work."

(2) and minor issue - end users do not know anything about technical details of the app, so the message text should ask them about clearing the form itself, not validation.

File: script.js [hunk 2]
```diff
@@ -9,13 +9,42 @@ const small_film_set = [
...
+        .then(() => $$("form").clearValidation())
```
Comment (roman-zaitsev) [inline script.js]: it can be located in one then

.then(() => {
     $$("form").clearValidation();
     $$("form").clear();
})

File: script.js [hunk 3]
```diff
@@ -53,25 +83,38 @@ const form = {
...
                 {
                     rows: [
                         { template: "EDIT FILMS", type: "section" },
-                        { view: "text", label: "Title" },
-                        { view: "text", label: "Year" },
-                        { view: "text", label: "Rating" },
-                        { view: "text", label: "Votes" },
+                        { view: "text", label: "Title", name: "title", invalidMessage: "Title can not be empty" },
+                        { view: "text", label: "Year", name: "year", invalidMessage: "Year should be between 1970 and current" },
+                        { view: "text", label: "Rating", name: "rating", invalidMessage: "Rating cannot be empty or 0" },
```
Comment (roman-zaitsev) [inline script.js]: here you can use 
**type: "number"**
to make it enter only numbers

File: script.js [hunk 4]
```diff
@@ -53,25 +83,38 @@ const form = {
...
+            rules: {
+                title: webix.rules.isNotEmpty,
+                year: function (value) {
+                    return value > 1970 && value < new Date().getFullYear();
+                },
+                votes: function (value) {
+                    return value < 100000;
+                },
+                rating: function (value) {
+                    return value != 0 && value != '';
```
Comment (roman-zaitsev) [inline script.js]: shorter version
**return +value;**

File: script.js (hunk 1 above)
Comment (AZhurav1ev) [reply script.js]: It’s very clear article, thank you. Already fixed, including text message.

File: script.js (hunk 2 above)
Comment (AZhurav1ev) [reply script.js]: Thanks for example.

File: script.js (hunk 3 above)
Comment (AZhurav1ev) [reply script.js]: Didn’t know about this type, thanks, will use it.

File: script.js (hunk 4 above)
Comment (AZhurav1ev) [reply script.js]: I agree, it’s much better.

**Pull request merged**